apps/studio/src/server/modules/user/user.router.ts

```diff
@@ -0,0 +1,282 @@
+import { TRPCError } from "@trpc/server"
+
+import {
+  countInputSchema,
+  countOutputSchema,
+  createInputSchema,
+  createOutputSchema,
+  deleteInputSchema,
+  deleteOutputSchema,
+  getUserInputSchema,
+  getUserOutputSchema,
+  hasInactiveUsersInputSchema,
+  hasInactiveUsersOutputSchema,
+  listInputSchema,
+  listOutputSchema,
+  updateInputSchema,
+  updateOutputSchema,
+} from "~/schemas/user"
+import { protectedProcedure, router } from "../../trpc"
+import { db, sql } from "../database"
+import { validatePermissionsForManagingUsers } from "../permissions/permissions.service"
+import {
+  createUser,
+  getUsersQuery,
+  validateEmailRoleCombination,
+} from "./user.service"
+
+export const userRouter = router({
+  create: protectedProcedure
+    .input(createInputSchema)
+    .output(createOutputSchema)
+    .mutation(async ({ ctx, input: { siteId, users } }) => {
+      await validatePermissionsForManagingUsers({
+        siteId,
+        userId: ctx.user.id,
+        action: "create",
+      })
+
+      const createdUsers = await db.transaction().execute(async (trx) => {
+        return await Promise.all(
+          users.map(async (user) => {
+            const { user: createdUser, resourcePermission } = await createUser({
+              ...user,
+              email: user.email.toLowerCase(),
+              siteId,
+              trx,
+            })
+            return {
+              id: createdUser.id,
+              email: createdUser.email,
+              role: resourcePermission.role,
+            }
+          }),
+        )
+      })
+
+      // TODO: Send welcome email to users
+      // Email service is not ready yet
```

Comment:
@adriangohjw will this be implemented in this sprint? It's labelled as a P1. Do heads up if we need to deprio this

Reply:
oops this comment was added before i worked on email service. Email service has since been done, i just need to pull the changes and integrate them (call the service endpoint)

Reply:
got it, thanks!!

Reply:
implemented in https://github.com/opengovsg/isomer/pull/1118/commits/d9ffcd67849db5dfb2c923fd12be8f52c0b99871